Marlin/src/gcode/motion/M290.cpp

```diff
@@ -40,7 +40,44 @@
 
 #if ENABLED(BABYSTEP_ZPROBE_OFFSET)
 
-  FORCE_INLINE void mod_probe_offset(const float &offs) {
+  FORCE_INLINE void apply_offset_steps(const uint8_t axis, const float &offs) {
+    if (axis == Z_AXIS && parser.boolval('P', true)) {
+      #if ENABLED(BABYSTEP_ZPROBE_OFFSET)
+        if (true
+          #if ENABLED(BABYSTEP_HOTEND_Z_OFFSET)
+            && active_extruder == 0
+          #endif
+        ) {
+          NOLESS(offs, Z_PROBE_OFFSET_RANGE_MIN - zprobe_zoffset);
+          NOMORE(offs, Z_PROBE_OFFSET_RANGE_MAX - zprobe_zoffset);
+        }
+        #if ENABLED(BABYSTEP_HOTEND_Z_OFFSET)
+          else if (offs > hotend_offset[active_extruder].z + hotend_offset_limit.z)
+            offs = -hotend_offset_limit.z - zprobe_zoffset;
+          else if (offs < hotend_offset[active_extruder].z - hotend_offset_limit.z)
+            offs = hotend_offset_limit.z - hotend_offset[active_extruder].z;
+        #endif
+        mod_zprobe_zoffset(offs);
+      #endif
+    }
+    #if ENABLED(BABYSTEP_XY) && EXTRUDERS > 1
+      else if (axis == X_AXIS) {
+        if (offs > hotend_offset[active_extruder].x + hotend_offset_limit.x)
+          offs = -hotend_offset_limit.x - zprobe_zoffset;
+        else if (offs < hotend_offset[active_extruder].x - hotend_offset_limit.x)
+          offs = hotend_offset_limit.x - hotend_offset[active_extruder].x;
+      }
+      else if (axis == Y_AXIS) {
+        if (offs > hotend_offset[active_extruder].y + hotend_offset_limit.y)
+          offs = -hotend_offset_limit.y - zprobe_zoffset;
+        else if (offs < hotend_offset[active_extruder].y - hotend_offset_limit.y)
+          offs = hotend_offset_limit.y - hotend_offset[active_extruder].y;
```

Comment:
I rearranged the comparisons to put `offs` in front of the compare operator. Please validate these changes.